xarray/core/dataarray.py

```diff
@@ -1736,5 +1736,93 @@ def dot(self, other):
 
         return type(self)(new_data, new_coords, new_dims)
 
+    def quantile(self, q, dim=None, axis=None, interpolation='linear'):
+        """
+
+        Compute the qth quantile of the data along the specified axis,
+        while ignoring nan values.
+        Returns the qth quantiles(s) of the array elements.
+
+        Parameters
+        ----------
+        q : float in range of [0,100] (or sequence of floats)
+            Quantile to compute, which must be between 0 and 100
+            inclusive.
+        dim : str or sequence of str, optional
+            Dimension(s) over which to apply quantile.
+        axis : int or sequence of int, optional
```

Comment:
I would not bother with the `axis` argument. It makes the logic below quite convoluted, and in my experience with xarray it's rarely useful.